scripts/minimal_init.lua

```diff
@@ -19,193 +9,379 @@ require("nvim-treesitter.configs").setup {
   ---- If you need to change the installation directory of the parsers (see -> Advanced Setup)
   -- Remember to run vim.opt.runtimepath:append("/some/path/to/store/parsers")!
   -- parser_install_dir = "/some/path/to/store/parsers",
+}
 
-  highlight = {
-    -- `false` will disable the whole extension
-    enable = true,
+vim.cmd [[
+packadd nvim-treesitter
+packadd nvim-treesitter-textobjects
+packadd plenary.nvim
+TSUpdate
+]]
 
-    -- Setting this to true will run `:h syntax` and tree-sitter at the same time.
-    -- Set this to `true` if you depend on 'syntax' being enabled (like for indentation).
-    -- Using this option may slow down your editor, and you may see some duplicate highlights.
-    -- Instead of true it can also be a list of languages
-    additional_vim_regex_highlighting = true,
+require("nvim-treesitter-textobjects").setup {
+  select = {
+    lookahead = true,
+    include_surrounding_whitespace = false,
   },
+  move = {
+    set_jumps = true,
+  },
+}
 
-  textobjects = {
-    select = {
-      enable = true,
+local select = require "nvim-treesitter-textobjects.select"
+for _, mode in ipairs { "x", "o" } do
+  vim.keymap.set(mode, "am", function()
```

Comment:
For `vim.keymap.set`, yes; but some of these functions have weird multiple dispatch. (I've removed such magic in nvim-treesitter; we should remove it here as well. Explicit is better.)

Reply:
@clason could you help me with a link to a PR/commit in nvim-treesitter with this kind of changes to get a better idea of how this should work?

Reply:
Not really, since I just removed everything ;) Here we need the mappings. This was just a general comment, though; if there's no magic and @ObserverOfTime's suggestion works, just do that.

Reply:
Or, conversely, if the function needs to be passed the `mode` because it can't work otherwise (or find out itself), then you need to keep the loop.